spec/message.abnf

```diff
@@ -32,6 +33,11 @@ literal-char = %x0-5B         ; omit \
              / %xE000-10FFFF
 
 variable = "$" name
+
+; reserve additional sigils for future use
+reserved = reserved-start text
+reserved-start = ("!" / "@" / "#" / "%" / "^" / "&" / "*" / "<" / ">" / "?" / "~")
```

Comment:
This is really cute, but it would disallow literal values that include `\{}`.

Currently this is a valid expression:
```
{:foo key=|{bar}|}
```
But if we were to unreserve `@` and try to give it the same semantics as with `:`, this would be a parse error as the expression would end before the literal does:
```
{@foo key=|{bar}|}
```

Leaving out the nesting from my proposal in https://github.com/unicode-org/message-format-wg/pull/371#discussion_r1155874966, I think we still need new `reserved-char` and `reserved-escape` definitions:

```suggestion
reserved = reserved-start reserved-body
reserved-start = "!" / "@" / "#" / "%" / "^" / "&" / "*" / "<" / ">" / "?" / "~"
reserved-body = *(reserved-char / reserved-escape / literal)
reserved-char = %x0-5B         ; omit \
              / %x5D-7A        ; omit { | }
              / %x7E-D7FF      ; omit surrogates
              / %xE000-10FFFF
reserved-escape = backslash ( backslash / "{" / "|" / "}" )
```